jib-core/src/main/java/com/google/cloud/tools/jib/image/json/JsonToImageTranslator.java

```diff
@@ -49,6 +49,15 @@
    */
   private static final Pattern portPattern = Pattern.compile("(\\d+)(?:/(tcp|udp))?");
 
+  /**
+   * Pattern used for parsing environment variables in the format {@code NAME=VALUE}. {@code NAME}
+   * should start with a letter and contain only letters (uppercase and lower) and numbers. {@code
+   * VALUE} can contain anything.
+   *
+   * <p>Example matches: NAME=VALUE, A12345=$$$$$
+   */
+  @VisibleForTesting static final Pattern environmentPattern = Pattern.compile("([^=].*?)=(.*)");
```

Comment:
This seems weird. `([^=].*?)` matches any string that doesn't start with `=`. I think it's conceptually correct to say `([^=]+)`.

Reply:
So as a concrete example,

```java
    Pattern pattern = Pattern.compile("([^=].*?)");
    Matcher matcher = pattern.matcher("ab==c==d");
    matcher.matches();
    System.out.println(matcher.group(1));
```

It prints out `ab==c==d`.

Reply:
Yeah I guess within the context of the rest of the expression, a name containing an `=` would only be detected if it started with an `=`. But yours makes more sense and is easier to read.